pkg/sharing/secret_exports.go

```diff
@@ -171,11 +180,58 @@ func (es exportedSecret) Matches(matcher SecretMatcher, nsIsExcludedFromWildcard
 			return false
 		}
 	}
+
 	if len(matcher.FromNamespace) > 0 {
 		if matcher.FromNamespace != es.secret.Namespace {
 			return false
 		}
 	}
+
+	selectors := es.export.Spec.ToSelectorMatchFields
```

Comment:
This is quite a big chunk of code. I would not be against moving this to a struct called NamespacesMatcher or something similar so that we can isolate this logic.

Reply:
Googled for json path lib and found gjson, didn't knew about already present lib `k8s.io/client-go/util/jsonpath`, thanks, changed to use `k8s.io/client-go/util/jsonpath` instead of gjson

Reply:
`NamespacesMatcher` struct implemented